### PR TITLE
mut_mut now more robust (thanks to Manishearth, see issue #9)

### PR DESCRIPTION
**Pull request by llogiq**

### CODE REVIEW

File: src/mut_mut.rs [hunk 1]
```diff
@@ -27,25 +23,34 @@ impl LintPass for MutMut {
...
 		}
 		
-		if unwrap_addr(expr).and_then(unwrap_addr).is_some() {
-			cx.span_lint(MUT_MUT, expr.span, 
-				"We're not sure what this means, so if you know, please tell us.")
-		}
+		unwrap_addr(expr).map(|e| {
+			if unwrap_addr(e).is_some() {
+				cx.span_lint(MUT_MUT, expr.span, 
+					"We're not sure what this means, so if you know, please tell us.")
```
Comment (Manishearth) [inline src/mut_mut.rs]: Perhaps a better message? "Generally you want to avoid `&mut &mut foo` if possible"

File: src/mut_mut.rs [hunk 2]
```diff
@@ -27,25 +23,34 @@ impl LintPass for MutMut {
...
+			} else {
+				match expr_ty(cx.tcx, e).sty {
+					ty_ptr(mt{ty: _, mutbl: MutMutable}) |
+					ty_rptr(_, mt{ty: _, mutbl: MutMutable}) => 
+						cx.span_lint(MUT_MUT, expr.span,
+							"This expression mutably borrows a mutable reference. Consider direct reborrowing"),
```
Comment (Manishearth) [inline src/mut_mut.rs]: nit: no "direct"

File: src/mut_mut.rs [hunk 3]
```diff
@@ -27,25 +23,34 @@ impl LintPass for MutMut {
...
-				"We're not sure what this means, so if you know, please tell us.")
-		}
+		unwrap_addr(expr).map(|e| {
+			if unwrap_addr(e).is_some() {
+				cx.span_lint(MUT_MUT, expr.span, 
+					"We're not sure what this means, so if you know, please tell us.")
+			} else {
+				match expr_ty(cx.tcx, e).sty {
+					ty_ptr(mt{ty: _, mutbl: MutMutable}) |
+					ty_rptr(_, mt{ty: _, mutbl: MutMutable}) =>
```
Comment (Manishearth) [inline src/mut_mut.rs]: not rptr, rptr is a raw pointer. `*mut *mut` is an okay pattern.

Comment (llogiq) [inline src/mut_mut.rs]: The rptr apparently matches line 14 of the test. Would you suggest this should not be an error?

Comment (Manishearth) [inline src/mut_mut.rs]: Sorry, `rptr` is a region pointer or a reference pointer (don't know what it stands for), `ptr` is raw. Oops.

Comment (Manishearth) [inline src/mut_mut.rs]: (so `ty_ptr` shouldn't be here)

**Pull request merged**